Zend/Optimizer/zend_inference.h

```diff
@@ -217,7 +217,7 @@ BEGIN_EXTERN_C()
 
 ZEND_API void zend_ssa_find_false_dependencies(const zend_op_array *op_array, zend_ssa *ssa);
 ZEND_API void zend_ssa_find_sccs(const zend_op_array *op_array, zend_ssa *ssa);
-ZEND_API int zend_ssa_inference(zend_arena **raena, const zend_op_array *op_array, const zend_script *script, zend_ssa *ssa, zend_long optimization_level);
+ZEND_API zend_result zend_ssa_inference(zend_arena **raena, const zend_op_array *op_array, const zend_script *script, zend_ssa *ssa, zend_long optimization_level);
```

Comment:
This should be fixed on PHP 8.3+. The size of enums is implementation defined, and as such might be smaller (or bigger) than `int`, so this might be ABI incompatible. However, the existing mismatch would already cause issues in that case, if the caller expects expects a `uint8_t` with the higher bits clobbered.

This looks safe, but I'm not an expert. It might be better to just avoid changing this until PHP 8.2 is phased out.

Reply:
ah ok I see, thanks

Reply:
I'm confused, why on 8.3+? Don't you mean to only do these changes on master?

Reply:
Nvm I get it, there's a mismatch in the .c file vs .h file, it's not a general remark.

Reply:
> This ~~should be~~ *is* fixed on PHP 8.3+

Sorry for the confusing wording.